src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldServer.java

```diff
@@ -75,4 +84,37 @@ public void onPostInit(CallbackInfoReturnable<World> ci) {
         IMixinWorld world = (IMixinWorld) ci.getReturnValue();
         world.updateWorldGenerator();
     }
+
+    @Overwrite
+    public void updateAllPlayersSleepingFlag() {
+        this.allPlayersSleeping = false;
+        if (!this.playerEntities.isEmpty()) {
+            int ignoredPlayers = 0;
+            int sleepingPlayers = 0;
+            for (EntityPlayer entityPlayer : this.playerEntities) {
+                if (entityPlayer.isSpectator()
+                        || (entityPlayer instanceof Player && ((Player)entityPlayer).isSleepingIgnored())) {
+                    ignoredPlayers++;
+                } else if (entityPlayer.isPlayerSleeping()) {
+                    sleepingPlayers++;
+                }
+            }
+            this.allPlayersSleeping = ((sleepingPlayers > 0)
+                    && (sleepingPlayers >= this.playerEntities.size() - ignoredPlayers));
+        }
+    }
+
+    @Overwrite
+    public boolean areAllPlayersAsleep() {
+        if ((this.allPlayersSleeping) && (!this.isRemote)) {
+            for (EntityPlayer entityPlayer : this.playerEntities) {
+                boolean ignore = entityPlayer instanceof Player && ((Player)entityPlayer).isSleepingIgnored();
+                if (!ignore && (entityPlayer.isSpectator() || !entityPlayer.isPlayerFullyAsleep())) {
```

Comment:
Just use `@Redirect` on the `isPlayerFullyAsleep()` call to check if the player's sleeping status is ignored. That might not be a valid override for every call to that method, but in this case it will be able to do the same task.